_drafts/2024-05-18-get-started-with-aws-tf-01.md

```diff
@@ -0,0 +1,254 @@
+---
+layout: post
+title: "Getting Started with AWS and Terraform: 01 - Creating an EC2 Intance"
+date: 2024-05-17 16:00:00 +0800
+categories: aws terraform infratructure IaC
+description: In this multi-part blog series we will learn how to create a modern enterprise ready web service in AWS using Terraform. In this first post we will create an EC2 instance to host our web service in AWS using Terraform
+---
+
+# Introduction
+
+In this series of blog posts, we will build an enterprise scale web infrastructure; Starting small with a simple virtual machine image, and iterating as our needs develop until eventually we will have a highly availble, secure, and resiliant web infrastructure capable of meeting the needs of tens of customers! All of this will be managed with the popular Infrastructure As Code tool, Terraform.  
+In this first post, we will:
+ 1. Configure Terraform with a pre-existing AWS account
+ 2. Deploy an EC2 instance to AWS
+ 3. Allow incoming connections to the EC2 instance from our local device
+ 4. SSH into our EC2 instance using a ssh key pair
+
+## Prerequisites
+
+Before we start building, you'll want to have setup a couple things:
+
+1. Have an [AWS Account](https://aws.amazon.com) setup
+2. Installed [Terraform cli](https://developer.hashicorp.com/terraform/install?product_intent=terraform)
+
+## Step 0 - Create a folder for your project
+
+As we'll be building our infractructure using Terraform, which is an IaC tool, we'll need somewhere to write this code.
```

Comment:
```suggestion
As we'll be building our infrastructure using Terraform, we'll need somewhere to write this code.
```

Small typo + removing some redundant (imo) information. We spoke briefly about figuring out what the assumed knowledge of your audience would be, unsure if you've defined that yourself but yeah, in my mind I think its safe to assume readers know Terraform is an IaC tool.